com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -707,6 +667,38 @@ private bool ProcessEvent()
         private void Update()
         {
             if (IsDisabledBySimulator)
+            {
+                // Flush all send queue items
+                foreach (var queueItem in m_SendQueue)
+                {
+                    queueItem.Value.Consume(queueItem.Value.Length);
+                }
+
+                // Flush all connection requests
+                while (m_Driver.IsCreated)
+                {
+                    var connection = m_Driver.Accept();
+
+                    if (connection == default)
+                    {
+                        break;
+                    }
+                }
+
+                // Flush all incoming events
+                while (m_Driver.IsCreated)
+                {
+                    var eventType = m_Driver.PopEvent(out var _, out var __, out var ___);
```

Comment:
Hmm, doesn't like there's a way to "peek" the queue?

Reply:
Unfortunately there's no way to peek at the queue. I'm not against adding the functionality, but I'm not sure how it would help here. It seems to me like disconnect events could simply be handled directly here (would just need to put the disconnect-handling code from `ProcessEvent` in its own method).